tests/core/test_core.py

```diff
@@ -126,13 +126,35 @@ def test_narrow_to_stream(
         id_list = index_stream["stream_msg_ids_by_stream_id"][stream_id]
         assert {widget.original_widget.message["id"]} == id_list
 
-    def test_narrow_to_topic(self, mocker, controller, msg_box, index_topic):
+    @pytest.mark.parametrize(
+        ["initial_narrow", "anchor"],
+        [
+            ([], None),
+            ([["stream", "PTEST"], ["topic", "Test"]], 537286),
```

Comment:
It's good to know that you've tested that this fails without the code change, but let's now move this line into the fixing commit, possibly with any extra test adjustment necessary to handle this (stream setting below?).

This test then becomes enhanced to handle positioning with multiple messages, and the next commit will extend it with code to pass the extra test case.